src/AspNet.Security.OAuth.Apple/Internal/DefaultAppleIdTokenValidator.cs

```diff
@@ -5,62 +5,67 @@
-using System.Text;
 using System.Threading.Tasks;
 using JetBrains.Annotations;
 using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.JsonWebTokens;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
 using Microsoft.IdentityModel.Tokens;
 
 namespace AspNet.Security.OAuth.Apple.Internal
 {
     internal sealed class DefaultAppleIdTokenValidator : AppleIdTokenValidator
     {
         private readonly ILogger _logger;
-        private readonly AppleKeyStore _keyStore;
-        private readonly CryptoProviderFactory _cryptoProviderFactory;
 
         public DefaultAppleIdTokenValidator(
-            [NotNull] AppleKeyStore keyStore,
-            [NotNull] CryptoProviderFactory cryptoProviderFactory,
             [NotNull] ILogger<DefaultAppleIdTokenValidator> logger)
         {
-            _keyStore = keyStore;
-            _cryptoProviderFactory = cryptoProviderFactory;
             _logger = logger;
         }
 
         public override async Task ValidateAsync([NotNull] AppleValidateIdTokenContext context)
         {
-            if (!context.Options.JwtSecurityTokenHandler.CanValidateToken)
+            if (context.Options.SecurityTokenHandler is null)
             {
-                throw new NotSupportedException($"The configured {nameof(JwtSecurityTokenHandler)} cannot validate tokens.");
+                throw new InvalidOperationException("The options SecurityTokenHandler is null.");
             }
 
-            byte[] keysJson = await _keyStore.LoadPublicKeysAsync(context);
+            if (!context.Options.SecurityTokenHandler.CanValidateToken)
+            {
+                throw new NotSupportedException($"The configured {nameof(JsonWebTokenHandler)} cannot validate tokens.");
+            }
 
-            string json = Encoding.UTF8.GetString(keysJson);
-            var keySet = JsonWebKeySet.Create(json);
+            if (context.Options.ConfigurationManager is null)
+            {
+                throw new InvalidOperationException($"An OpenID Connect configuration manager has not been set on the {nameof(AppleAuthenticationOptions)} instance.");
+            }
 
-            var parameters = new TokenValidationParameters()
+            if (context.Options.TokenValidationParameters is null)
             {
-                CryptoProviderFactory = _cryptoProviderFactory,
-                IssuerSigningKeys = keySet.Keys,
-                ValidAudience = context.Options.ClientId,
-                ValidIssuer = context.Options.TokenAudience,
-            };
+                throw new InvalidOperationException($"Token validation parameters have not been set on the {nameof(AppleAuthenticationOptions)} instance.");
+            }
+
+            OpenIdConnectConfiguration configuration = await context.Options.ConfigurationManager.GetConfigurationAsync(context.HttpContext.RequestAborted);
+
+            context.Options.TokenValidationParameters.IssuerSigningKeys = configuration.JsonWebKeySet.Keys;
```

Comment:
Assuming this method can be concurrently called, this is a risky operation. Instead, you may want to clone `context.Options.TokenValidationParameters` and mutate the cloned instance.